app/queries/get_grouped_content_and_links.rb

```diff
@@ -0,0 +1,116 @@
+module Queries
+  class GetGroupedContentAndLinks
+    PAGE_SIZE = 10
+    MAX_PAGE_SIZE = 1000
+    DEFAULT_CONTENT_ID = '00000000-0000-0000-0000-000000000000'.freeze
+
+    include ActiveModel::Validations
+
+    attr_accessor :last_seen_content_id, :page_size
+
+    validate :has_valid_uuid_for_pagination
+    validates_numericality_of :page_size, less_than_or_equal_to: MAX_PAGE_SIZE # FIXME: > 0
+
+    def initialize(last_seen_content_id: nil, page_size: nil)
+      @last_seen_content_id = last_seen_content_id || DEFAULT_CONTENT_ID
+      @page_size = page_size || PAGE_SIZE
+    end
+
+    # Returns an array of hashes where each hash contains a content_id, an
+    # array of content item rows, and an array of link rows.
+    def call
+      content_ids = query_content_ids_for_page(
+        last_seen_content_id: @last_seen_content_id,
+        page_size: @page_size
+      )
+
+      group_results(content_results(content_ids), link_set_results(content_ids))
+    end
+
+    def has_valid_uuid_for_pagination
+      unless @last_seen_content_id == DEFAULT_CONTENT_ID || UuidValidator.valid?(@last_seen_content_id)
```

Comment:
It just makes for a friendlier API.

If the request is made with an invalid UUID, then the client is doing something wrong, and we could tell them why.

In this case though it's inconsistent with the other requests so I'll take it out. This means we will return 404 in all cases.